doc/vector/code-samples/zvkg.s

```diff
@@ -51,3 +51,92 @@ zvkg_vghsh:
     vghsh.vv v0, v8, v4
     vse32.v v0, (a0)
     ret
+
+
+# zvkg_vghsh_vv
+#
+# Performs vector add-multiply over GHASH Galois-Field for multiple
+# 128 bit elements groups. 'n' is the number of 128b groups.
+# The input arrays should be 32b aligned on processors that do not
+# support unaligned 32b vector loads/stores.
+#     Y_out = ((Y_in ^ X) o H)
+#
+#   void zvkg_vghsh_vv(
+#       uint32_t* Y,   // a0
+#       uint32_t* X,   // a1
+#       uint32_t* H,   // a2
+#       size_t    n    // a3
+#   );
+#
+.balign 4
+.global zvkg_vghsh_vv
+zvkg_vghsh_vv:
+    beqz a3, 2f  # Early exit in the "0 bytes to process" case
+    # a3 on input is number of 128b groups in the input arrays. We multiply
+    # by 4 as the Zvk instructions expect VSEW=32. a3 becomes the number
+    # of 32b elements to process, which is a multiple of 4.
+    slli a3, a3, 2
+1:
+    # We use LMUL=4 to enable runs with VLEN=32, as a proof of concept.
+    # Once VLEN>=128, we can simply use LMUL=1.
+    vsetvli t0, a3, e32, m4, ta, ma
+
+    vle32.v v0, (a0)
+    vle32.v v4, (a1)
+    vle32.v v8, (a2)
+    vghsh.vv v0, v8, v4  # Y(v0) = Y(v0) ^ X(v4)) o H(v8)
+    vse32.v v0, (a0)
+
+    sub a3, a3, t0       # Decrement number of remaining 32b elements
+    slli t0, t0, 2       # t0 (#bytes consumed) <- t0 (#4B) * 4
+    add a0, a0, t0
+    add a1, a1, t0
+    add a2, a2, t0
+    bnez a3, 1b          # More elements to process?
+
+2:
+    ret
+
+# zvkg_vgmul_vv
+#
+# Performs vector multiply over GHASH Galois-Field for multiple
+# 128 bit elements groups.
+# 'n' is the number of 128b element groups, n operations will be performed.
+# The input arrays should be 32b aligned on processors that do not
+# support unaligned 32b vector loads/stores.
+#
+#     Vd_out = (Vd_in o Vs2)
+#
+#   void zvkg_vgmul_vv(
+#       uint32_t* vd,    // a0
+#       uint32_t* vs2,   // a1
+#       size_t    n      // a2
+#   );
+#
```

Comment:
Why not keep the same labels as in the header file ? (`zvkg_vgmul_vv` could have even use `Y` and  `H`)
```suggestion
#     X_out = (X_in o Y)
#
#   void zvkg_vgmul_vv(
#       uint32_t* X,    // a0
#       uint32_t* Y,   // a1
#       size_t    n      // a2
#   );
#
```

Reply:
Done.